uclient-fetch.c

```diff
@@ -633,6 +670,26 @@ int main(int argc, char **argv)
 			case L_QUIET:
 				quiet = true;
 				break;
+			case L_HEADER:
+				tmp = strchr(optarg, ':');
+				if (!tmp)
+					return usage(progname);
+
+				*tmp++ = '\0';
+				while (isspace(*tmp))
+					++tmp;
+
+				if (*tmp == '\0' || !is_valid_header(optarg) || strchr(tmp, '\n'))
+					return usage(progname);
+
+				h = calloc(1, sizeof(*h));
+				if (!h)
+					exit(ENOMEM);
```

Comment:
The exit is used incorrectly here. It should be just `exit(1)`.

The ENOMEM is a value of a globa variable errno that is set by the calloc.
You can print the `perror` function:

```c
perror("headers");
exit(1);
```
And on the OOM it will print: `headers: Cannot allocate memory`.

You can pragmatically set limit to test it's behavior:
https://stackoverflow.com/a/9154138/1049542